Model/Checkout/PlaceOrder.php

```diff
@@ -236,9 +237,16 @@ public function createOrder(Quote $quote, string $amwalOrderId, string $refId):
     {
         $this->logDebug(sprintf('Submitting quote with ID %s', $quote->getId()));
         $order = $this->getOrderByAmwalOrderId($amwalOrderId);
-        if (!$order->getEntityId()) {
+
+        if ($order->getEntityId()) {
+            if ($order->getState() == Order::STATE_PENDING_PAYMENT) {
+                $this->orderRepository->delete($order);
```

Comment:
Should we really delete these or cancel them instead? Deleting is a lot more impactful, and means you will lose part of the sales history

Reply:
The issue is that one of the customers made more than one order for the same AmwalOrderId, and Cornjob then edits all the orders that have the AmwalOrderId, so we now had 3 orders that had one AmwalOrderId, so our fix we deleted the order and created it again with the same AmwalOrderId without duplicate any orders, also if any calculation issues happend will fix in the new order.